tests/components/mqtt/test_water_heater.py

```diff
@@ -0,0 +1,1008 @@
+    help_test_entity_device_info_update,
+    help_test_entity_device_info_with_connection,
+    help_test_entity_device_info_with_identifier,
+    help_test_entity_id_update_discovery_update,
+    help_test_entity_id_update_subscriptions,
+    help_test_publishing_with_custom_encoding,
+    help_test_reloadable,
+    help_test_setting_attribute_via_mqtt_json_message,
+    help_test_setting_attribute_with_template,
+    help_test_setting_blocked_attribute_via_mqtt_json_message,
+    help_test_unique_id,
+    help_test_unload_config_entry_with_platform,
+    help_test_update_with_json_attrs_bad_json,
+    help_test_update_with_json_attrs_not_dict,
+)
+
+from tests.common import async_fire_mqtt_message
+from tests.components.water_heater import common
+from tests.typing import MqttMockHAClientGenerator, MqttMockPahoClient
+
+ENTITY_WATER_HEATER = "water_heater.test"
+
+
+DEFAULT_CONFIG = {
+    mqtt.DOMAIN: {
+        water_heater.DOMAIN: {
+            "name": "test",
+            "mode_command_topic": "mode-topic",
+            "away_mode_command_topic": "away-mode-topic",
+            "temperature_command_topic": "temperature-topic",
+            "temperature_low_command_topic": "temperature-low-topic",
+            "temperature_high_command_topic": "temperature-high-topic",
+        }
+    }
+}
+
+
+@pytest.fixture(autouse=True)
+def water_heater_platform_only():
+    """Only setup the water heater platform to speed up tests."""
+    with patch("homeassistant.components.mqtt.PLATFORMS", [Platform.WATER_HEATER]):
+        yield
+
+
+@pytest.mark.parametrize("hass_config", [DEFAULT_CONFIG])
+async def test_setup_params(
+    hass: HomeAssistant, mqtt_mock_entry: MqttMockHAClientGenerator
+) -> None:
+    """Test the initial parameters."""
+    await mqtt_mock_entry()
+
+    state = hass.states.get(ENTITY_WATER_HEATER)
+
+    assert state.attributes.get("temperature") == 110
+    assert state.state == "off"
+    assert state.attributes.get("min_temp") == DEFAULT_MIN_TEMP
+    assert state.attributes.get("max_temp") == DEFAULT_MAX_TEMP
+
+
+@pytest.mark.parametrize("hass_config", [DEFAULT_CONFIG])
+async def test_supported_features(
+    hass: HomeAssistant, mqtt_mock_entry: MqttMockHAClientGenerator
+) -> None:
+    """Test the supported_features."""
+    await mqtt_mock_entry()
+
+    state = hass.states.get(ENTITY_WATER_HEATER)
+    support = (
+        WaterHeaterEntityFeature.TARGET_TEMPERATURE
+        | WaterHeaterEntityFeature.OPERATION_MODE
+        | WaterHeaterEntityFeature.AWAY_MODE
+    )
+
+    assert state.attributes.get("supported_features") == support
+
+
+@pytest.mark.parametrize("hass_config", [DEFAULT_CONFIG])
+async def test_get_operation_modes(
+    hass: HomeAssistant, mqtt_mock_entry: MqttMockHAClientGenerator
+) -> None:
+    """Test that the operation list returns the correct modes."""
+    await mqtt_mock_entry()
+
+    state = hass.states.get(ENTITY_WATER_HEATER)
+    modes = state.attributes.get("operation_list")
```

Comment:
No need to define modes, you can just do: assert [...] == state.attributes.get("operation_list")

Reply:
Good point, thanks!